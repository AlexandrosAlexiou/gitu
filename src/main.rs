use std::{
    error::Error,
    io::{stderr, BufWriter},
};

use clap::Parser;
use crossterm::{
    terminal::{disable_raw_mode, enable_raw_mode, EnterAlternateScreen, LeaveAlternateScreen},
    ExecutableCommand,
};
use ratatui::{prelude::CrosstermBackend, Terminal};

pub fn main() -> Result<(), Box<dyn Error>> {
    let mut terminal = Terminal::new(CrosstermBackend::new(BufWriter::new(stderr())))?;
    terminal.hide_cursor()?;
    enable_raw_mode()?;
    stderr().execute(EnterAlternateScreen)?;

    let result = gitu::run(gitu::cli::Args::parse(), &mut terminal);

    stderr().execute(LeaveAlternateScreen)?;
    disable_raw_mode()?;

    result?;

    Ok(())
}
