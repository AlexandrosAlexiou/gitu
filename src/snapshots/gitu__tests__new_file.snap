---
source: src/main.rs
expression: "redact_hashes(terminal, dir)"
---
Buffer {
    area: Rect { x: 0, y: 0, width: 60, height: 20 },
    content: [
        " No branch                                                  ",
        "                                                            ",
        "ðŸ¢’Untracked files                                            ",
        " new-file                                                   ",
        "                                                            ",
        " Recent commits                                             ",
        "                                                            ",
        "                                                            ",
        "                                                            ",
        "                                                            ",
        "                                                            ",
        "                                                            ",
        "                                                            ",
        "                                                            ",
        "                                                            ",
        "                                                            ",
        "                                                            ",
        "                                                            ",
        "                                                            ",
        "                                                            ",
    ],
    styles: [
        x: 0, y: 0, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 0, y: 2, fg: Reset, bg: Rgb(80, 73, 69), underline: Reset, modifier: NONE,
        x: 1, y: 2, fg: Rgb(216, 166, 87), bg: Rgb(80, 73, 69), underline: Reset, modifier: BOLD,
        x: 16, y: 2, fg: Reset, bg: Rgb(80, 73, 69), underline: Reset, modifier: NONE,
        x: 0, y: 3, fg: Reset, bg: Rgb(42, 40, 39), underline: Reset, modifier: NONE,
        x: 1, y: 3, fg: Rgb(234, 105, 98), bg: Rgb(42, 40, 39), underline: Reset, modifier: BOLD,
        x: 9, y: 3, fg: Reset, bg: Rgb(42, 40, 39), underline: Reset, modifier: NONE,
        x: 0, y: 4, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
        x: 1, y: 5, fg: Rgb(216, 166, 87), bg: Reset, underline: Reset, modifier: BOLD,
        x: 15, y: 5, fg: Reset, bg: Reset, underline: Reset, modifier: NONE,
    ]
}
